core/coreapi/name_test.go

```diff
@@ -21,45 +22,150 @@ func addTestObject(ctx context.Context, api coreiface.CoreAPI) (coreiface.Path,
 	return api.Unixfs().Add(ctx, files.NewReaderFile("", "", ioutil.NopCloser(&io.LimitedReader{R: rnd, N: 4092}), nil))
 }
 
-func TestBasicPublishResolve(t *testing.T) {
-	ctx := context.Background()
-	nds, apis, err := makeAPISwarm(ctx, true, 5)
+func appendPath(path coreiface.Path, sub string) coreiface.Path {
+	p, err := coreiface.ParsePath(path.String() + sub)
```

Comment:
Even though it will result in the same behaviour, I wonder if we should make a point to use `path.Join` in these instances.